admin-frontend/open_admin_app/lib/widgets/feature-groups/environment_drop_down.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:bloc_provider/bloc_provider.dart';
+import 'package:flutter/material.dart';
+import 'package:mrapi/api.dart';
+import 'package:open_admin_app/widgets/feature-groups/feature-groups-bloc.dart';
+
+class EnvironmentDropDown extends StatefulWidget {
+  final FeatureGroupsBloc bloc;
+  final String? envId;
+  const EnvironmentDropDown({Key? key, required this.bloc, this.envId})
+      : super(key: key);
+
+  @override
+  State<EnvironmentDropDown> createState() => _EnvironmentDropDownState();
+}
+
+class _EnvironmentDropDownState extends State<EnvironmentDropDown> {
+  String? _selectedEnvId;
+
+  @override
+  void initState() {
+    super.initState();
+
+    final bloc = BlocProvider.of<FeatureGroupsBloc>(context);
+
+    // when application changes, this stream will set appropriate ID or null
+    bloc.currentEnvironmentStream.listen((env) {
```

Comment:
you need to hold onto the stream subscription and un-listen when  the finalise happens for stateful objects.